src/index.ts

```diff
@@ -21,6 +21,10 @@ export function extract(
       });
     });
 
+  if (options?.withDuplicateMessage) {
+    return { messages: sort(messages), errors };
```

Comment:
It is better to avoid using the same action in different lines of code (here and on line 36, at the end of this function). As a solution, you can invert the condition, and put the existing code inside it:

```
if (!options?.withDuplicateMessage) {
  const ids = Object.keys(messages);
  ...
}
```

Reply:
+ it will be great to see some tests for this new option :)